textproto/header.go

```diff
@@ -396,6 +421,19 @@ func trimAroundNewlines(v []byte) string {
 	return b.String()
 }
 
+const (
+	MaxHeaderLines = 1000
+
+	// MaxLineOctets is the maximum length of line in the header in bytes (octets).
+	//
+	// The max length in RFC 5322 is 1000 **charcters** (including CRLF).
```

Comment:
It's not clear what a "character" is, because RFC 5322 only considers US-ASCII. Also RFC 5322 is for e-mails specifically, not for MIME messages in general. I agree 4k is a reasonable limit though.

Reply:
I thought https://tools.ietf.org/html/rfc6532#section-3.4 redefines the limit to be 998 Unicode characters, but apparently it is still in octets.  Since the reasoning is invalid, I guess I will just drop it and round the limit up to 4000.

Reply:
Additional note: this limit is per line, not per header field. So the RFC doesn't forbid very long header fields.